SEGV: "super compile -dag -C" on query that reads known Parquet

## tl;dr

```
$ super -f parquet -o f.parquet -c "values {foo:1}" &&
  super compile -dag -C "SELECT * FROM f.parquet"

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xd5ab423]

goroutine 1 [running]:
github.com/brimdata/super/compiler/semantic.(*analyzer).fileScanColumns(0xc000031be0, 0xde55078?)
	/Users/phil/work/super/compiler/semantic/op.go:158 +0x123
github.com/brimdata/super/compiler/semantic.(*analyzer).semFromEntity(0xc000031be0, {0xde5d660?, 0xc0002f4e70}, 0x0, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/semantic/op.go:104 +0x609
github.com/brimdata/super/compiler/semantic.(*analyzer).semFromElem(0xc000031be0, 0xc00068e3c0, {0x0?, 0x1?, 0xc000030dd0?})
	/Users/phil/work/super/compiler/semantic/op.go:57 +0x5b
github.com/brimdata/super/compiler/semantic.(*analyzer).semFrom(0xd126047?, 0xc0000e6600?, {0x0?, 0xc000030eb0?, 0xbe9e985?})
	/Users/phil/work/super/compiler/semantic/op.go:47 +0x3b
github.com/brimdata/super/compiler/semantic.(*analyzer).semSelectFrom(0xc000031be0, {0x0, 0x16}, 0xdac0b60?, {0x0?, 0x0, 0xc000699398?})
	/Users/phil/work/super/compiler/semantic/sql.go:288 +0x51
github.com/brimdata/super/compiler/semantic.(*analyzer).semSelect(0xc000031be0, 0xc0003a83c0, {0x0?, 0xc000031068?, 0xbf0c9e5?})
	/Users/phil/work/super/compiler/semantic/sql.go:33 +0x6f
github.com/brimdata/super/compiler/semantic.(*analyzer).semSQLOp(0xc000031be0, {0xde5dd20?, 0xc0003a83c0}, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/semantic/sql.go:470 +0x57a
github.com/brimdata/super/compiler/semantic.(*analyzer).semOp(0xc000031be0, {0xde5dd20?, 0xc0003a83c0}, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/semantic/op.go:554 +0x20d0
github.com/brimdata/super/compiler/semantic.(*analyzer).semSeq(0xc000031be0, {0xc0003961a0, 0x1, 0x1})
	/Users/phil/work/super/compiler/semantic/op.go:37 +0x51
github.com/brimdata/super/compiler/semantic.Analyze({0xde65500, 0xc0006b4720}, 0xc00036af80, 0xc00036afa0, 0x0)
	/Users/phil/work/super/compiler/semantic/analyzer.go:25 +0x325
github.com/brimdata/super/compiler.Analyze(...)
	/Users/phil/work/super/compiler/package.go:27
github.com/brimdata/super/cmd/super/compile.(*Shared).Run(0xc000386908, {0xde655a8, 0xc00038ad20}, {0xc00003c090?, 0xd72ee90?, 0x13?}, 0xbef4071?, 0x0, 0x0)
	/Users/phil/work/super/cmd/super/compile/shared.go:81 +0x2b9
github.com/brimdata/super/cmd/super/compile.(*Command).Run(0xc000386900, {0xc00003c090, 0x1, 0x1})
	/Users/phil/work/super/cmd/super/compile/command.go:57 +0xd7
github.com/brimdata/super/pkg/charm.path.run({0xc000396a00, 0x2, 0x2}, {0xc00003c090, 0x1, 0x0?})
	/Users/phil/work/super/pkg/charm/path.go:11 +0x7b
github.com/brimdata/super/pkg/charm.(*Spec).Exec(0xefaaf40, {0xc00003c060, 0x4, 0x4})
	/Users/phil/work/super/pkg/charm/charm.go:74 +0x1fa
main.main()
	/Users/phil/work/super/cmd/super/main.go:39 +0x5b
```

## Details

Repro is with super commit 0413b1f.

I don't often reach for `super compile -dag -C`, but after seeing @mccanne recently show its output to a community user I figured I should try to make it part of my debug workflow.

The SEGV shown above doesn't happen if I write the Parquet data into a file without extension and let auto detect do its thing.

```
$ super -version
Version: 0413b1fa1

$ super -f parquet -o f -c "values {foo:1}" &&
  super compile -dag -C "SELECT * FROM f"

file f
| values {in:this}
| values {in:in,out:{...in}}
| values out
| output main
```

Though it does SEGV if I use that same file without extension but include the `(FORMAT parquet)`.

```
$ super -f parquet -o f -c "values {foo:1}" &&
  super compile -dag -C "SELECT * FROM f (FORMAT parquet)"

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x32ea423]

goroutine 1 [running]:
github.com/brimdata/super/compiler/semantic.(*analyzer).fileScanColumns(0xc00052dbe0, 0x3b94078?)
	/Users/phil/work/super/compiler/semantic/op.go:158 +0x123
...
```

--- a/cmd/super/compile/shared.go
+++ b/cmd/super/compile/shared.go
@@ -77,7 +77,7 @@ func (s *Shared) Run(ctx context.Context, args []string, dbFlags *dbflags.Flags,
 		return s.writeValue(ctx, ast.Parsed())
 	}
 	rctx := runtime.DefaultContext()
-	env := exec.NewEnvironment(nil, root)
+	env := exec.NewEnvironment(storage.NewLocalEngine(), root)
 	dag, err := compiler.Analyze(rctx, ast, env, extInput)
 	if err != nil {
 		return err

